src/versionbits.cpp

```diff
@@ -210,6 +205,16 @@ class VersionBitsConditionChecker : public AbstractThresholdConditionChecker {
 
 protected:
     int64_t BeginTime(const Consensus::Params& params) const override { return params.vDeployments[id].nStartTime; }
+    int MasternodeBeginHeight(const Consensus::Params& params) const override {
+        const auto& deployment = params.vDeployments[id];
+        if (deployment.nMNActivationHeight == 0) {
+            return std::numeric_limits<int>::max();
+        }
+        if (deployment.nMNActivationHeight < 0) {
+            return 0;
+        }
```

Comment:
This looks redundant and even a bit confusing imo.

Reply:
that's legit condition. For case `if nMNActivationHeight < 0` it means that this fork is activated not as MN EHF (dip 23), but regular hard-fork. In this case Mn Acivation Height equals to 0 -> it means that it can be activated by from height of 0 by extra conditions from DIP23

For case `if MNActivation Height == 0` it means that this fork is activated by MN EHF, but signal is not mined yet -> minimal allowed height equals to "max int".

For case if `MN activation height` is positive number it means a minimal height when it can be activated.

Should I refactor this part to make it more sense?